ros_buildfarm/templates/release/rpm/sourcepkg_job.xml.em

```diff
@@ -126,6 +126,18 @@ but disabled since the package is blacklisted (or not whitelisted) in the config
     'builder_shell',
     script='\n'.join([
         'echo "# BEGIN SECTION: Upload sourcerpm"',
+        'find sourcepkg -mindepth 1 -maxdepth 1 -type f -name "*.src.rpm" -fprint sourcepkg/rpm_upload_args.txt -fprintf sourcepkg/rpm_import_args.txt "--import=/tmp/upload-${BUILD_TAG}/%f\\n"',
+        'ssh %s -- mkdir -p /tmp/upload-${BUILD_TAG}/' % (upload_host,),
+        'xargs -a sourcepkg/rpm_upload_args.txt -I @ scp @ %s:/tmp/upload-${BUILD_TAG}/' % (upload_host,),
+        'xargs -a sourcepkg/rpm_import_args.txt ssh %s -- createrepo-agent /var/repos/%s/building/%s/' % (upload_host, os_name, os_code_name),
+        'ssh %s -- rm -fr /tmp/upload-${BUILD_TAG}/' % (upload_host,),
+        'echo "# END SECTION"',
```

Comment:
This is similar enough to the binarypkg import (which has additional options for invalidation) to make me wonder if it's worth factoring this script section into a snippet.

Mostly so that we don't have to remember to change both if we tweak something. I think it's discretionary though as piling more options onto a snippet may make it even harder to read through the python string templating and shell scripting.

Reply:
I gave this an honest shot, but it's up to 5 template variables now and is starting to get hard to read. I'd like to revisit the upload process with some optimizations later - maybe a script somewhere will make sense and reduce the duplication.

Reply:
That's more effort than I anticipated certainly! Thanks for giving it a serious attempt and my apologies for sending you down a rabbit hole.